### PR TITLE
Fix empty content

### PR DESCRIPTION
**Pull request by vvalgenti**
Fixes an issue that could arise if a snort rule has no content options.

### CODE REVIEW

File: sniffles/ruletrafficgenerator.py [hunk 1]
```diff
@@ -1586,11 +1586,15 @@ def generate_full_eval(self, rule=None):
...
 
-    def generate_nfa_data(self, rule=None):
+    def generate_nfa_data(self, rule=None, length=0):
         if rule:
             data = []
             http_content = []
             content_options = rule.getContent()
+            if content_options is None:
+                if length == -1:
+                    length = random.randint(0, 1400) + 10
```
Comment (msk) [inline sniffles/ruletrafficgenerator.py]: Why not `randint(10, 1410)`?

File: sniffles/ruletrafficgenerator.py [hunk 2]
```diff
@@ -1586,11 +1586,15 @@ def generate_full_eval(self, rule=None):
                     path = []
                     self.follow_all_branches(nfa, nfa.start, path)
 
-    def generate_nfa_data(self, rule=None):
+    def generate_nfa_data(self, rule=None, length=0):
```
Comment (msk) [inline sniffles/ruletrafficgenerator.py]: Wouldn't `-1` (random length) be more reasonable default value? It's more likely that "no argument" in calling "generate_nfa_data" means "Just generate data; I don't care how many." than "Do not generate data".

File: sniffles/ruletrafficgenerator.py [hunk 3]
```diff
@@ -1586,11 +1586,15 @@ def generate_full_eval(self, rule=None):
...
         if rule:
             data = []
             http_content = []
             content_options = rule.getContent()
+            if content_options is None:
+                if length == -1:
```
Comment (msk) [inline sniffles/ruletrafficgenerator.py]: I think `None` represents "no restriction on length" better than `-1`.

Comment (vvalgenti) [reply sniffles/ruletrafficgenerator.py]: The system is currently built around the idea of -1 implying random.  In fact, I believe it is even possible to designate a range using negatives (-10 is a range between 1 and 10).  I cannot remember for sure (that was present in at least on iteration).  However, changing from -1 has potentially far reaching changes.  If we want to make this change it is best to do it as a separate card.

Comment (msk) [reply sniffles/ruletrafficgenerator.py]: OK. Then let's make the task a separate issue.

Comment (vvalgenti) [reply sniffles/ruletrafficgenerator.py]: Task created.

File: sniffles/ruletrafficgenerator.py (hunk 1 above)
Comment (vvalgenti) [reply sniffles/ruletrafficgenerator.py]: completed.

Comment (msk) [reply sniffles/ruletrafficgenerator.py]: I don't see any change.

**Pull request merged**